### PR TITLE
[SNMP] Greenlets to speed-up snmp walk

### PR DESCRIPTION
**Pull request by truthbk**
Description sums it up nicely, we put in place a greenlet pool to try to improve scheduling while grabbing the snmp results.

### CODE REVIEW

File: requirements-opt.txt [hunk 1]
```diff
@@ -39,3 +39,7 @@ psycopg2==2.6
 # It's a pure python module, it doesn't require anything to install,
 # but needs the pywin32 extension to work
 wmi==1.4.9
+
+# checks.d/snmp/pu
+# Require gevent
+gevent==1.0.1
```
Comment (yannmh) [inline requirements-opt.txt]: Why not using the last stable release, i.e [`1.0.2`](https://github.com/gevent/gevent/blob/1.0.2/changelog.rst#release-102-may-23-2015) ?

Comment (yannmh) [inline requirements-opt.txt]: Don't forget to reflect this add-on on [dd-agent-omnibus](https://github.com/DataDog/dd-agent-omnibus/blob/d0ade317b5956ef17c6ead007655a0c65cdafa11/config/projects/datadog-agent.rb#L136-L154) and [omnibus-software](https://github.com/DataDog/omnibus-software/tree/master/config/software) :wink: .

**Comment by remh** [comment]
Any reason not to use the threadpool we are already using for network checks ?

**Comment by truthbk** [comment]
I'm actually going to close this for now and give it another spin because I'm not seeing the expected performance gains - will take a look at the thread pool.

**Comment by truthbk** [comment]
@remh thanks for the input, pretty decent improvement for multiple instances by converting it into a network check.

**Comment by truthbk** [comment]
Greenlets not playing well with threads (event after disabling monkey patching for threads) at the moment so closing in favor of https://github.com/DataDog/dd-agent/pull/2152

**Pull request closed without merging**